### PR TITLE
160 layout bugs

### PR DESCRIPTION
**Pull request by michaelthatsit**
lot's of changes, updating readme as well. might rename mr-container.

### CODE REVIEW

File: src/UI/UIEntity.js [hunk 1]
```diff
@@ -59,6 +59,35 @@ export class MRUIEntity extends Entity {
...
+    add(entity) {
+        let container = this.closest('mr-panel');
+
+        if (container && entity instanceof MRUIEntity) {
+            container.add(entity);
+        } else {
+            this.object3D.add(entity.object3D);
+        }
+
+        entity.object3D.position.z += 0.0001;
```
Comment (hanbollar) [inline src/UI/UIEntity.js:75]: needs comment above this line for why this bump is necessary

File: src/UI/media/Image.js [hunk 2]
```diff
@@ -36,6 +36,8 @@ export class MRImage extends MRUIEntity {
             }
         });
         this.object3D.material.map = this.texture;
+
+        this.object3D.position.z += 0.00001;
```
Comment (hanbollar) [inline src/UI/media/Image.js:40]: needs comment above this line for why this bump is necessary

File: src/component-systems/TextSystem.js [hunk 3]
```diff
@@ -63,15 +63,18 @@ export class TextSystem extends System {
                     entity.blur();
                 }
             } else {
-                text = entity.textContent.replace(/(\n)\s+/g, '$1').trim();
+                text = entity.textContent
+                    .replace(/(\n)\s+/g, '$1')
+                    .replace(/(\r\n|\n|\r)/gm, '')
+                    .trim();
```
Comment (hanbollar) [inline src/component-systems/TextSystem.js:69]: need comment above this explaining why this swap is needed and safe

File: src/core/entity.js [hunk 4]
```diff
@@ -49,8 +50,9 @@ export default class Entity extends MRElement {
      *
      */
     get height() {
-        const styleHeight = this.compStyle.height.split('px')[0] > 0 ? this.compStyle.height.split('px')[0] : window.innerHeight;
-        return (styleHeight / window.innerHeight) * global.viewPortHeight;
+        const rect = this.getBoundingClientRect();
+        // const styleHeight = this.compStyle.height.split('px')[0] > 0 ? this.compStyle.height.split('px')[0] : window.innerHeight;
```
Comment (hanbollar) [inline src/core/entity.js:54]: can this line be deleted?

File: src/entities/Model.js [hunk 5]
```diff
@@ -15,9 +15,9 @@ export class Model extends Entity {
     /**
      *
      */
-    get height() {
-        return this.contentHeight;
-    }
+    // get height() {
+    //     return this.contentHeight;
+    // }
```
Comment (hanbollar) [inline src/entities/Model.js:20]: can the above be deleted?

File: src/entities/Surface.js [hunk 6]
```diff
@@ -35,10 +48,12 @@ export class Surface extends Entity {
      *
      */
     connected() {
-        this.windowVerticalScale = this.height / 3;
-        this.windowHorizontalScale = this.width / 3;
+        // this.clipping = new ClippingGeometry(new THREE.BoxGeometry(this.width, this.height, 0.3));
```
Comment (hanbollar) [inline src/entities/Surface.js:51]: can this line be deleted?

File: src/entities/Surface.js [hunk 7]
```diff
@@ -35,10 +48,12 @@ export class Surface extends Entity {
...
+        this.windowVerticalScale = this.height; /// 3;
+        this.windowHorizontalScale = this.width; // / 3;
```
Comment (hanbollar) [inline src/entities/Surface.js:53]: question for understanding: (i see this was moved to line 67) is the divide by 3 just a choice or is there a reason we're using 3 specifically?

File: src/entities/Surface.js [hunk 8]
```diff
@@ -111,18 +126,24 @@ export class Surface extends Entity {
         this.object3D.position.copy(this.anchorPosition);
         this.object3D.quaternion.copy(this.anchorQuaternion);
 
+        console.log('floating', this.floating);
+
+        if (!this.floating) {
+            this.rotationPlane.rotation.x = (3 * Math.PI) / 2;
```
Comment (hanbollar) [inline src/entities/Surface.js:132]: need comment as to why this additional rotation is needed

**Review by hanbollar — APPROVED** [review]
some quick notes for comments / straggler lines and a question or two 🎉

File: README.md [hunk 9]
```diff
@@ -43,98 +43,54 @@ Both options require you generate an ssl certificate & key via openssl:
...
-                    <mr-img src="..."></mr-img>
-                    <mr-row height="0.02">
-                        <mr-button onClick="Prev()"> <- </mr-button>
-                        <mr-button onClick="Next()"> -> </mr-button>
-                    </mr-row>
-                </mr-column>
-            </mr-row>
-        </mr-container>
-    </mr-surface>
+    <!-- The 2D UI Panel -->
```
Comment (hanbollar) [inline README.md:80]: could we also put one element that is 3d outside of the panel for the readme demo? ik we have included the mr-model elemnt on line 88, just a bit concerned of using just the naming of 2D though

**Review by hanbollar — APPROVED** [review]
one quick additional question

File: src/entities/Surface.js (hunk 7 above)
Comment (michaelthatsit) [reply src/entities/Surface.js:53]: > question for understanding: (i see this was moved to line 67) is the divide by 3 just a choice or is there a reason we're using 3 specifically?

This is a choice atm, the window is really big if we stick to 1:1 scale. I'd like to refactor the anchoring/surface system overall. so we can parameterize this in the future?

File: README.md (hunk 9 above)
Comment (michaelthatsit) [reply README.md:80]: ooo good catch. I might add this in another PR with the button/img update. I'd like it to be in the demo page as well.

File: src/entities/Surface.js (hunk 7 above)
Comment (hanbollar) [reply src/entities/Surface.js]: sounds good - then in line 67, either add a comment or can we separate the divide by 3 into a constant first and then divide - better visual separation for future parameterization - just dont want it to end up as a 'secret special number'

Comment (michaelthatsit) [reply src/entities/Surface.js]: Made it a global.

**Comment by michaelthatsit** [comment]
@hanbollar good to merge?

**Pull request merged**